lib/src/main/scala/spinal/lib/MasterSlave.scala

```diff
@@ -37,6 +42,11 @@ trait IMasterSlave {
     _isMasterInterface = Some(false)
   }
 
+  final def setAsMonitor(): Unit = {
+    asMonitor()
+    _isMasterInterface = Some(false)
```

Comment:
This is saying "I am a slave", so the types extending this trait can assume that. It could lead to unexpected behavior because it is not really a slave. Maybe it should not be declared as master nor slave because it isn't any of them?

```suggestion
```